cli/src/pcluster/config/cluster_config.py

```diff
@@ -856,6 +856,71 @@ def __init__(self, head_node_bootstrap_timeout: int = None, compute_node_bootstr
         )
 
 
+class BudgetNotification(Resource):
+    """Represent the configuration of each notification under the NotificationsWithSubscribers field of a budget."""
+
+    def __init__(
+        self,
+        notification_type: str = None,
+        comparison_operator: str = None,
+        threshold: int = None,
+        threshold_type: str = None,
+    ):
+        super().__init__()
+        self.notification_type = Resource.init_param(notification_type, default="ACTUAL")
+        self.comparison_operator = Resource.init_param(comparison_operator, default="GREATER_THAN")
+        self.threshold = Resource.init_param(threshold)
+        self.threshold_type = Resource.init_param(threshold_type, default="ABSOLUTE_VALUE")
+
+
+class BudgetSubscriber(Resource):
+    """Represent the configuration of an individual subscriber of a budget notification."""
+
+    def __init__(self, subscription_type: str = None, address: str = None):
+        super().__init__()
+        self.subscription_type = Resource.init_param(subscription_type, default="EMAIL")
+        self.address = Resource.init_param(address)
+
+
+class BudgetNotificationWithSubscribers(Resource):
+    """Represent the configuration of the NotificationWithSubscribers field of a budget."""
+
+    def __init__(self, notification: BudgetNotification = None, subscribers: List[BudgetSubscriber] = None):
+        super().__init__()
+        self.notification = notification
+        self.subscribers = subscribers
+
+
+class BudgetLimit(Resource):
+    """Represent the configuration of a budget limit."""
+
+    def __init__(self, amount: int = None, unit: str = None):
```

Comment:
Please double check the amount type here

Reply:
Done!